config/main.py

```diff
@@ -1761,13 +1859,22 @@ def startup(ctx, interface_name):
 def shutdown(ctx, interface_name):
     """Shut down interface"""
     log_info("'interface shutdown {}' executing...".format(interface_name))
-    config_db = ctx.obj['config_db']
+    namespace = ctx.obj['namespace']
+    if  sonic_device_util.is_multi_npu() and namespace is None:
+        ns = get_intf_namespace(interface_name)
+        if ns is None:
+            ctx.fail("namespace [-n] option required to configure interface {}".format(interface_name))
+        else:
+            namespace = ns
+    config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+    config_db.connect()
```

Comment:
remove code duplication.

1822, 1862, ...

Reply:
Will remove code duplication, will make it functions and invoke those, thanks.

Reply:
Defined new function and used it to remove the code duplicated.